src/main/java/org/broadinstitute/consent/http/service/UserFileService.java

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.consent.http.service;
+
+import com.google.cloud.storage.BlobId;
+import org.broadinstitute.consent.http.cloudstore.GCSService;
+import org.broadinstitute.consent.http.db.UserFileDAO;
+import org.broadinstitute.consent.http.enumeration.UserFileCategory;
+import org.broadinstitute.consent.http.models.UserFile;
+
+import javax.ws.rs.NotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+public class UserFileService {
+
+    GCSService gcsService;
+    UserFileDAO userFileDAO;
+
+    public UserFileService(UserFileDAO userFileDAO, GCSService gcsService) {
+        this.userFileDAO = userFileDAO;
+        this.gcsService = gcsService;
+    }
+
+    UserFile uploadAndStoreUserFile(
+            InputStream content,
+            String fileName,
+            String mediaType,
+            UserFileCategory category,
+            String entityId,
+            Integer createUserId
+    ) throws IOException {
+        // upload to GCS
+        BlobId blobId = gcsService.storeDocument(
+                content,
+                mediaType,
+                UUID.randomUUID().toString());
+
+        // insert file
+        Integer userFileId = userFileDAO.insertNewFile(
+                fileName,
+                category.getValue(),
+                blobId.getBucket(),
+                blobId.getName(),
+                mediaType,
+                entityId,
+                createUserId,
+                new Date()
+        );
+
+        return userFileDAO.findFileById(userFileId);
+    }
+
+    // fetches file from GCS and adds it to the userFile
+    private void fetchAndPopulateUploadedFile(UserFile userFile) throws NotFoundException  {
+        InputStream document = gcsService.getDocument(
+                BlobId.of(
+                        userFile.getBucketName(),
+                        userFile.getBlobName()));
+
+        userFile.setUploadedFile(document);
+    }
+
+    public UserFile fetchById(
+            Integer userFileId
+    ) throws NotFoundException {
+        UserFile userFile = userFileDAO.findFileById(userFileId);
+        // download file from GCS
+        fetchAndPopulateUploadedFile(userFile);
```

Comment:
I'm not sold on fetching stream content _ALL THE TIME_. We should really only be pulling GCS content when a user specifically requests it. Imagine this user flow ... 

1. User navigates to an entity.
2. User should see that entity has associated files
  a. User then navigates away from page (no need to ever have queried for the IS content)
  b. User clicks on an associated file entity (now we need to pull the IS content

We should have two methods, one to get the file as metadata and a separate one to get the content.

Reply:
Yes I agree. I envisioned a `findFile` method as well. I can implement it as well.

Reply:
@rushtong -could it be even simpler as a passthrough to GCSService.getDocument and return the InputStream?

Reply:
> could it be even simpler as a passthrough to GCSService.getDocument and return the InputStream?

Yes, that is the intention of that service call.